.github/workflows/upload.yml

```diff
@@ -0,0 +1,54 @@
+name: Upload release
+
+on:
+  # https://docs.github.com/en/webhooks/webhook-events-and-payloads#release
+  release:
+    types: [published]
+
+concurrency:
+  # Concurrency group that uses the workflow name and PR number if available
+  # or commit SHA as a fallback. If a new build is triggered under that
+  # concurrency group while a previous build is running it will be canceled.
+  # Repeated pushes to a PR will cancel all previous builds, while multiple
+  # merges to main will not cancel.
+  group: ${{ github.workflow }}-${{ github.ref_name || github.sha }}
+  cancel-in-progress: true
+
+permissions:
+  contents: write
+
+jobs:
+  # create source archive and upload it to the published release
+  # URL to the archive: https://github.com/conda/<repo>/releases/download/<tag>/<repo>-<tag>.tar.gz
+  upload:
+    if: !github.event.repository.fork
```

Comment:
we need this to be quoted since it starts with a `!`: https://results.pre-commit.ci/run/github/6235174/1709882115.GVlSQPmNRfmcGfUd4aFBWQ

Reply:
Oh woops, my IDE yelled at me telling me I'm wrong, will undo